programs/compressed-token/src/burn.rs

```diff
@@ -27,13 +34,86 @@ pub struct CompressedTokenInstructionDataBurn {
     pub delegated_transfer: Option<DelegatedTransfer>,
 }
 
+#[derive(Accounts)]
+pub struct BurnInstruction<'info> {
+    #[account(mut)]
+    pub fee_payer: Signer<'info>,
+    pub authority: Signer<'info>,
+    /// CHECK: that mint authority is derived from signer
+    #[account(seeds = [CPI_AUTHORITY_PDA_SEED], bump,)]
+    pub cpi_authority_pda: UncheckedAccount<'info>,
+    /// CHECK: that authority is mint authority
+    #[account(mut, constraint = mint.mint_authority.unwrap() == authority.key())]
```

Comment:
`constraint = mint.mint_authority.unwrap() == authority.key()`
'is this required by spl burn?